File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.go
func characterReplacement(s string, k int) int {
    freq := map[byte]int{}
    maxFreq, sol := 0, 0
    for l, r := 0, 0; r < len(s); r++ {
        freq[s[r]] = freq[s[r]]+1
        maxFreq = max(maxFreq, freq[s[r]])

        for (r-l+1) - maxFreq > k {
            freq[s[l]]--
            l++
        }

        sol = max(sol, (r-l+1));
    }
    return sol
}

File: Data Structures & Algorithms/partition-labels/submission-0.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        Map<Character, Integer> indices = new HashMap<>();

        for (int i = 0; i < s.length(); i++) indices.put(s.charAt(i), i);

        List<Integer> sol = new ArrayList<>();
        for (int i = 0, l = 0, r = 0; i < s.length(); i++) {
            r = Math.max(r, indices.get(s.charAt(i)));
            if (i == r) {
                sol.add(r - l + 1);
                l = i+1;
            }
        }
        return sol;
    }
}
